tests/foreman/cli/test_bootstrap_script.py

```diff
@@ -15,62 +15,205 @@
 
 :Upstream: No
 """
-from robottelo.decorators import stubbed, tier1, upgrade
+from nailgun import entities
+from robottelo.decorators import stubbed, tier1, tier4, upgrade
+from robottelo import manifests
+from robottelo.api.utils import (
+    promote, enable_rhrepo_and_fetchid, upload_manifest
+    )
 from robottelo.test import CLITestCase
+from robottelo.config import settings
+from robottelo.containers import Container
+from robottelo.constants import (
+    DEFAULT_ARCHITECTURE,
+    RHEL_7_MAJOR_VERSION,
+    REPOS,
+    PRDS,
+    REPOSET,
+)
 
 
 class BootstrapScriptTestCase(CLITestCase):
     """Test class for bootstrap script."""
 
     @classmethod
     def setUpClass(cls):
-        """create VM for testing """
+        """Set up organization and location for tests."""
         super(BootstrapScriptTestCase, cls).setUpClass()
-
-    @tier1
-    @stubbed()
+        cls.org = entities.Organization().create()
+        cls.loc = entities.Location(organization=[cls.org]).create()
+
+        # Clone manifest to get RHEL7 subs
+        with manifests.clone() as manifest:
+            upload_manifest(cls.org.id, manifest.content)
+        rh_repo_id = enable_rhrepo_and_fetchid(
+            basearch='x86_64',
+            org_id=cls.org.id,
+            product=PRDS['rhel'],
+            reposet=REPOSET['rhel7'],
+            repo=REPOS['rhel7']['name'],
+            releasever='7Server',
+        )
+        rh_repo = entities.Repository(id=rh_repo_id).read()
+
+        # Create essentials for an activation key
+        lce = entities.LifecycleEnvironment(organization=cls.org.id).create()
+        cv = entities.ContentView(
+            organization=cls.org.id,
+            repository=[rh_repo],
+        ).create()
+        cv.publish()
+        cv = cv.read()
+        promote(cv.version[0], environment_id=lce.id)
+        cv = cv.read()
+
+        # Create an activation key and set it to not auto-attach
+        cls.ak = entities.ActivationKey(
+            environment=cls.org.library,
+            auto_attach=False, content_view=cv, organization=cls.org.id,
+            ).create()
+
+        prod = rh_repo.product.read()
+        subs = entities.Subscription().search(
+            query={'search': 'name~{0}'.format(prod.name)}
+        )
+
+        cls.ak.add_subscriptions(data={'subscriptions': [{'id': subs[0].id}]})
+
+        # Search for SmartProxy
+        proxy = entities.SmartProxy().search(
+            query={
+                u'search': u'name={0}'.format(
```

Comment:
You can drop this `u` prefix in new code. It was there only to force Python2 to treat string as unicode. Python3 does that by default. And we no longer pretend we support Python2.

Reply:
OK, thank you